platform/kube/inject/inject.go

```diff
@@ -137,46 +140,71 @@ func injectIntoPodTemplateSpec(p *Params, t *v1.PodTemplateSpec) error {
 	t.Annotations["pod.beta.kubernetes.io/init-containers"] = string(initAnnotationValue)
 
 	// sidecar proxy container
-	t.Spec.Containers = append(t.Spec.Containers,
-		v1.Container{
-			Name:  proxyContainerName,
-			Image: p.ProxyImage,
-			Args: []string{
-				"proxy",
-				"sidecar",
-				"-s", p.ManagerAddr,
-				"-m", p.MixerAddr,
-				"-n", "$(POD_NAMESPACE)",
-				"-v", strconv.Itoa(p.Verbosity),
+	args := []string{
+		"proxy",
+		"sidecar",
+		"-s", p.ManagerAddr,
+		"-m", p.MixerAddr,
+		"-n", "$(POD_NAMESPACE)",
+		"-v", strconv.Itoa(p.Verbosity),
+	}
+	var volumeMounts []v1.VolumeMount
+	if p.EnableAuth {
+		args = append(args, "--enable_auth", "--auth_config_path", p.AuthConfigPath)
+		volumeMounts = append(volumeMounts, v1.VolumeMount{
+			Name:      istioCertVolumeName,
+			ReadOnly:  true,
+			MountPath: p.AuthConfigPath,
+		})
+
+		sa := t.Spec.ServiceAccountName
+		if sa == "" {
+			sa = "default"
```

Comment:
I believe K8S uses the default service account unless it's otherwise specified in the pod spec.